### PR TITLE
Scan fragment

### PR DESCRIPTION
**Pull request by KelvinKappeler**
Implementation of the scan fragment and its tests.
This fragment allows you to take a picture and (for the moment) just display it with a text explaining if the operation was successful.

### CODE REVIEW

File: app/src/main/java/com/github/bjolidon/bootcamp/utils/Compatibility.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     * Returns the value associated with the given key or null if no mapping of the desired type exists for the given key.
+     */
+    fun <T> getDataActivityResult(activityResult: ActivityResult, key: String, clazz: Class<T>): T? {
+        return if (android.os.Build.VERSION.SDK_INT >= 33) {
+            activityResult.data?.extras?.getParcelable(key, clazz)
+        } else {
+            clazz.cast(activityResult.data?.extras?.get(key))
+        }
+    }
+}
```
Comment (Cocograal) [inline app/src/main/java/com/github/bjolidon/bootcamp/utils/Compatibility.kt:22]: add end line

File: app/src/main/res/layout/fragment_scanner.xml [hunk 2]
```diff
@@ -7,17 +7,52 @@
...
+        android:layout_marginTop="32dp"
+        android:contentDescription="@string/scanner_content_description_image"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.495"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/text_information"
+        app:layout_constraintVertical_bias="0.318"
+        tools:src="@tools:sample/avatars" />
 </androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (Cocograal) [inline app/src/main/res/layout/fragment_scanner.xml:58]: add end line

**Review by Cocograal — CHANGES_REQUESTED** [review]
Overall the code is good, just do not forget the end lines

File: app/src/main/java/com/github/bjolidon/bootcamp/ui/scanner/ScannerFragment.kt [hunk 3]
```diff
@@ -1,41 +1,53 @@
...
-  }
 
-override fun onDestroyView() {
+    override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
-}
\ No newline at end of file
+}
```
[the remark this answers is not in the document]
Comment (KelvinKappeler) [reply app/src/main/java/com/github/bjolidon/bootcamp/ui/scanner/ScannerFragment.kt:53]: Already there

**Review by Cocograal — APPROVED** [review]
All is good now!

File: app/src/main/java/com/github/bjolidon/bootcamp/ui/scanner/ScannerFragment.kt [hunk 4]
```diff
@@ -1,41 +1,53 @@
...
 import androidx.lifecycle.ViewModelProvider
 import com.github.bjolidon.bootcamp.databinding.FragmentScannerBinding
+import com.github.bjolidon.bootcamp.utils.Compatibility
 
+/**
+ * This fragment is responsible to take a picture of the card
+ */
 class ScannerFragment : Fragment() {
+    private var _binding: FragmentScannerBinding? = null
+    private val binding get() = _binding!!
```
Comment (SolalPirelli) [inline app/src/main/java/com/github/bjolidon/bootcamp/ui/scanner/ScannerFragment.kt:21]: It looks like this is (still) unused and could be removed?

**Pull request merged**